lib/swaps/errors.ts

```diff
@@ -15,6 +16,10 @@ const errors = {
     message: `swapClient for currency ${currency} is not configured`,
     code: errorCodes.SWAP_CLIENT_NOT_CONFIGURED,
   }),
+  SECRET_HASH_NOT_FOUND: (hash: string) => ({
```

Comment:
As far as naming goes, "not found" sounds like we're unable to find that secret hash internally, instead of it being unspecified on a request that we receive. Also the "secret hash" terminology seems exclusive to raiden, everywhere else we are using `rHash` and in fact there is a check in the `Service.resolveHash()` method to ensure that `rHash` has a value (so it would catch the times that it's not present). Having this in `swaps/errors.ts` seems slightly misplaced.

In #999 you make a raiden error called `INVALID_TOKEN_PAYMENT_RESPONSE` for when the secret hash is missing. Maybe you can use a raiden error called `INVALID_RESOLVE_REQUEST` here?

Reply:
@sangaman you're right - the naming for this was bad. Just to clarify - this is not checking the inputs from Raiden, just the output.

I changed it to `SWAP_DEAL_NOT_FOUND` and moved the error throwing logic into `Swaps`. I still feel like `swaps/errors` is the correct place for this because it's now being thrown from `Swaps.resolveHash`. I think the "not found" terminology is correct because Raiden requests a resource (secret) from xud. If we cannot find the resource it makes sense to respond with 404 not found.

Reply:
Ok it makes much more sense now, maybe I misunderstood earlier. Ironically tho, I now think that something more generic like `PAYMENT_HASH_NOT_FOUND` might be more appropriate, since we use this error for sanity swaps as well which don't have "deal" involved per se. Just a suggestion, definitely not a deal breaker, and otherwise this all looks good.

Reply:
Updated the error to `PAYMENT_HASH_NOT_FOUND`.